src/solvers/refinement/string_refinement.cpp

```diff
@@ -2047,7 +2047,7 @@ static bool is_linear_arithmetic_expr(const exprt &expr)
 ///   false otherwise.
 static bool universal_only_in_index(const string_constraintt &expr)
 {
-  for(auto it=expr.depth_begin(); it!=expr.depth_end();)
+  for(auto it=expr.body().depth_begin(); it!=expr.body().depth_end();)
```

Comment:
Could you please squash this with the previous commit? I still have doubts that it is correct. Can we look at it together?